junk/gen-bench-data/src/main.rs

```diff
@@ -735,6 +751,257 @@ fn big_values_data() -> Data {
     }
 }
 
+fn generate_random_data(random_data_size: usize) -> Vec<u8> {
+    use rand::thread_rng;
+    use rand::Rng;
+
+    // hex::encode later prints out every byte as 2 bytes sequence.
+    let random_data_size = random_data_size / 2 * MB;
+
+    let mut random_data = Vec::<u8>::with_capacity(random_data_size);
+    let mut rng = thread_rng();
```

Comment:
It makes the benches unreproducible (even if no variability is expected).  I suggest using a fixed seed.